File: src/UIs/BlazorWebApp/FoodDelivery.BlazorWebApp/Dtos/UpdateProductRequest.cs
namespace FoodDelivery.BlazorWebApp.Dtos;

public class UpdateProductRequest
{
    public string Name { get; set; } = default!;
    public decimal Price { get; set; }
    public int Stock { get; set; }
    public int RestockThreshold { get; set; }
    public int MaxStockThreshold { get; set; }
    public int Status { get; set; }
    public int ProductType { get; set; }
    public int Color { get; set; }
    public int Height { get; set; }
    public int Width { get; set; }
    public int Depth { get; set; }
    public string Size { get; set; } = default!;
    public long CategoryId { get; set; }
    public long SupplierId { get; set; }
    public long BrandId { get; set; }
    public string? Description { get; set; }
}

File: src/UIs/Spa/react-food-delivery/src/features/products/dtos/update-product-request.ts
export interface UpdateProductRequest {
    name: string;
    price: number;
    stock: number;
    restockThreshold: number;
    maxStockThreshold: number;
    status: number;
    productType: number;
    color: number;
    height: number;
    width: number;
    depth: number;
    size: string;
    categoryId: number;
    supplierId: number;
    brandId: number;
    description?: string | null;
}
